torch/_inductor/models/mm_kernel_prediction_model.py

```diff
@@ -0,0 +1,678 @@
+        )
+
+    def vec(
+        self, m: int, n: int, k: int, dsize: int, config: Any
+    ) -> tuple[int, int, int, int, int, int, int, int, int]:
+        """
+        Convert matrix multiplication parameters and config to a feature vector.
+
+        Args:
+            m: First dimension of matrix multiplication
+            n: Second dimension of matrix multiplication
+            k: Third dimension of matrix multiplication
+            dsize: Data size in bits (e.g., 16 for float16, 32 for float32)
+            config: Configuration object containing kernel parameters
+
+        Returns:
+            Tuple containing the extracted features
+        """
+        kwargs = config.all_kwargs()
+
+        return (
+            int(m),
+            int(n),
+            int(k),
+            int(dsize),
+            int(kwargs["BLOCK_M"]),
+            int(kwargs["BLOCK_N"]),
+            int(kwargs["BLOCK_K"]),
+            int(kwargs["num_stages"]),
+            int(kwargs["num_warps"]),
+        )
+
+    @staticmethod
+    def vec_params(
+        m: int, n: int, k: int, dsize: int, params: TritonGEMMConfig
+    ) -> tuple[int, int, int, int, int, int, int, int, int]:
+        """
+        Convert matrix multiplication parameters and config to a feature vector.
+
+        Args:
+            m: First dimension of matrix multiplication
+            n: Second dimension of matrix multiplication
+            k: Third dimension of matrix multiplication
+            dsize: Data size in bits (e.g., 16 for float16, 32 for float32)
+            config: Configuration object containing kernel parameters
+
+        Returns:
+            Tuple containing the extracted features
+        """
+
+        return (
+            int(m),
+            int(n),
+            int(k),
+            int(dsize),
+            int(params.block_m),
+            int(params.block_n),
+            int(params.block_k),
+            int(params.num_stages),
+            int(params.num_warps),
+        )
+
+    def encode(
+        self, m: int, n: int, k: int, dtype: torch.dtype, configs: list[Any]
+    ) -> torch.Tensor:
+        """
+        Encode the matrix multiplication parameters and configs as input tensors for the model.
+
+        Args:
+            m: First dimension of matrix multiplication
+            n: Second dimension of matrix multiplication
+            k: Third dimension of matrix multiplication
+            dtype: Data type of the matrices
+            configs: List of configuration objects
+
+        Returns:
+            Tensor containing the encoded inputs ready for the model
+
+        Raises:
+            ValueError: If the dtype is not supported
+        """
+        # Determine data size based on dtype
+        if dtype == torch.bfloat16 or dtype == torch.float16:
+            dsize = 16
+        elif dtype == torch.float32:
+            dsize = 32
+        else:
+            raise ValueError(f"Unsupported dtype: {dtype}. Add support for this dtype.")
```

Comment:
```suggestion
        dsize = dtype.itemsize * 8
```

(I'm not sure how this plays into the broader logic; it might still be good to check that it's a float and is 16- or 32-bit)